pkg/bootstrap/docker/dockermachine/helper.go

```diff
@@ -147,10 +150,32 @@ func Client(name string) (*docker.Client, error) {
 		client, err = docker.NewVersionedClient(dockerHost, "")
 	}
 	if err != nil {
-		return nil, errors.NewError("could not get Docker client for machine %s", name).WithCause(err)
+		return nil, nil, errors.NewError("could not get Docker client for machine %s", name).WithCause(err)
 	}
 	client.SkipServerVersionCheck = true
-	return client, nil
+
+	var httpClient *http.Client
+	if len(certPath) > 0 {
```

Comment:
docker code uses len(certPath) to decide whether to create the tlscOptions. I would say that's the right one. We should update the first one.